openqaoa/optimizers/training_vqa.py

```diff
@@ -297,7 +297,7 @@ def results_dictionary(self,
         date_time = datetime.now().strftime("%d.%m.%Y_%H.%M.%S")
         file_name = f'opt_results_{date_time}' if file_name is None else file_name
         
-        self.qaoa_result = Result(self.log, self.method, self.vqa.cost_hamiltonian)
+        self.qaoa_result = Result(self.log, self.method, self.vqa.cost_hamiltonian, type(self.vqa))
```

Comment:
So, result expects a `backend` variable which declared as a string. However, _in general With one argument, return the type of an object. The return value is a type object and generally the same object as returned by `object.__class__`_  (see  https://docs.python.org/3/library/functions.html#type)

Could this eventually be a problem? It may be worth double checking

Reply:
shouldn't be declared as a string (changed). It shouldn't be a problem, but if someone changes  `QAOABaseBackendStatevector.__class__` should also change line 136, otherwise tests will fail